### DISCUSSION TITLE
Fuzzy matching - does not match keywords in the middle

### DISCUSSION THREAD

**Discussion by zsmatrix62** [hrsh7th/nvim-cmp#1440] [Q&A] [answered]
Hi,

I have this scenario: my object `payload` has an attribute `HCDetials` say. I got suggestion by `cmp` correctly if I type `HC..` or lowercased `hc ..`

<img width="392" alt="image" src="https://user-images.githubusercontent.com/2951879/218683459-517bb10b-d2e1-4d69-a1a5-b9d0c87a40fc.png">

But it does not work if I type something in the middle of the word, e.g `tail`:

<img width="516" alt="image" src="https://user-images.githubusercontent.com/2951879/218683684-1d78ecbb-611d-4a74-82aa-0b1f0dcfb1f3.png">

My settings have already been set for fuzzy match I think, does this require any other tweaks? Thanks.

<img width="446" alt="image" src="https://user-images.githubusercontent.com/2951879/218683810-72ad90a6-2426-4c30-9656-69a87cc0c142.png">

**Comment by hrsh7th** [comment, marked as answer]
LSP has an "isIncomplete" function for completion.

As such, the language server may not respond to `tail` query inputs.

It's depending on to the language server. What language server do you use?

**Reply by zsmatrix62** [reply]
in above screenshot is [tsserver](https://www.npmjs.com/package/typescript-language-server) .. hmm, so there must be something in the lsp setup?

**Reply by hrsh7th** [reply]
It seems to not bug of the language server side.

I'm trying to install luajit and luarocks via homebrew but I can't manage it... so it takes a time maybe.

**Reply by zsmatrix62** [reply]
It's okay,  thanks for taking note of this thread.  maybe we can finding out the reason in several weeks. :D

**Reply by hrsh7th** [reply]
Fixed https://github.com/hrsh7th/nvim-cmp/commit/6635b7aee56a47b0debe20e65195a1ca3b19e403

**Reply by zsmatrix62** [reply]
Thanks!  It works the fuzzy match. however, I type 'tails' but it higlited `H...ails`, does this normal?

<img width="272" alt="image" src="https://user-images.githubusercontent.com/2951879/218767433-3324c6fc-3041-489c-8e5b-29e969f9c72b.png">

**Reply by hrsh7th** [reply]
No... it's bug. I'll check it.

**Reply by hrsh7th** [reply]
re fixed. Thank you.

**Reply by zsmatrix62** [reply]
it works perfectly, thanks!
